### PR TITLE
Eli/editions v2

### PR DESCRIPTION
**Pull request by EGuenz**

### CODE REVIEW

**Review by michaeljklein — APPROVED** [review]
LGTM

**Comment by EGuenz** [comment]
https://better-call.dev/edo2net/KT1EYC1mn2fcxwY39vX9zVdUhqHy9q2zxLq1/

**Review by smondet — APPROVED** [review]
LGTM, I know don't what is the blessed way of doing that in Ligo, but it'd be nice to centralize the token-id-arithmetic in one place.

E.g. Instead of having deep in the code:

```ocaml
let edition_id = token_id / storage.max_editions_per_collection in
```

having inlined-functions or macros, something like:

```
#define TOKEN_ID_TO_EDITION_ID(storage, token_id) (token_id / storage.max_editions_per_collection)
```

File: packages/minter-contracts/bin/editions_token_metadata_view.tz [hunk 1]
```diff
@@ -1 +1 @@
-{ UNPAIR ; SWAP ; DUP ; DUG 2 ; CDR ; CAR ; CDR ; CDR ; CDR ; SWAP ; DUP ; DUG 2 ; GET ; IF_NONE { DROP 2 ; PUSH string "FA2_TOKEN_UNDEFINED" ; FAILWITH } { DUP ; CDR ; PUSH string "edition_id" ; GET ; IF_NONE { SWAP ; DROP ; SWAP ; DROP } { SWAP ; DROP ; UNPACK nat ; IF_NONE { DROP 2 ; PUSH string "FA2_TOKEN_UNDEFINED" ; FAILWITH } { DIG 2 ; CAR ; CDR ; SWAP ; GET ; IF_NONE { DROP ; PUSH string "FA2_TOKEN_UNDEFINED" ; FAILWITH } { CDR ; CAR ; SWAP ; PAIR } } } } } 
\ No newline at end of file
+{ UNPAIR ; SWAP ; DUP ; DUG 2 ; CDR ; CDR ; CAR ; CDR ; CAR ; SWAP ; DUP ; DUG 2 ; GET ; IF_NONE { DROP 2 ; PUSH string "FA2_TOKEN_UNDEFINED" ; FAILWITH } { DROP ; SWAP ; DUP ; DUG 2 ; CAR ; CDR ; SWAP ; DUP ; DUG 2 ; EDIV ; IF_NONE { PUSH string "DIV by 0" ; FAILWITH } {} ; CAR ; DIG 2 ; CAR ; CAR ; SWAP ; GET ; IF_NONE { DROP ; PUSH string "FA2_TOKEN_UNDEFINED" ; FAILWITH } { CDR ; CAR ; SWAP ; PAIR } } }
```
[the remark this answers is not in the document]
Comment (EGuenz) [reply packages/minter-contracts/bin/editions_token_metadata_view.tz:1]: > https://better-call.dev/edo2net/KT1EYC1mn2fcxwY39vX9zVdUhqHy9q2zxLq1/

I used the old view in this one by accident. This one has the correct view https://tzcomet.io/#/token/KT1PLERpGZJkKbXtBEN6jGL4jqERDWJKY3RN/1%3Fexplorer-input%3DKT1PLERpGZJkKbXtBEN6jGL4jqERDWJKY3RN

[the remark this answers is not in the document]
Comment (emishur) [reply packages/minter-contracts/bin/editions_token_metadata_view.tz:1]: TZComent shows `KT1PLERpGZJkKbXtBEN6jGL4jqERDWJKY3RN` `token_metadata` view as impure.
It looks like view implementation is pure (in TZIP-16 interpretation)

[the remark this answers is not in the document]
Comment (EGuenz) [reply packages/minter-contracts/bin/editions_token_metadata_view.tz]: thanks fixed

**Comment by EGuenz** [comment]
> LGTM, I know don't what is the blessed way of doing that in Ligo, but it'd be nice to centralize the token-id-arithmetic in one place.
> 
> E.g. Instead of having deep in the code:
> 
> ```ocaml
> let edition_id = token_id / storage.max_editions_per_collection in
> ```
> 
> having inlined-functions or macros, something like:
> 
> ```
> #define TOKEN_ID_TO_EDITION_ID(storage, token_id) (token_id / storage.max_editions_per_collection)
> ```

Thanks I inlined it.

File: packages/minter-contracts/bin/editions_token_metadata_view.tz (hunk 1 above)
[the remark this answers is not in the document]
Comment (EGuenz) [reply packages/minter-contracts/bin/editions_token_metadata_view.tz]: https://tzcomet.io/#/token/KT1S6CShPP79LWrHyLxkJ8prL7hRuQ8u6193/0%3Fexplorer-input%3DKT1S6CShPP79LWrHyLxkJ8prL7hRuQ8u6193

https://better-call.dev/edo2net/KT1S6CShPP79LWrHyLxkJ8prL7hRuQ8u6193/operations

**Pull request merged**